services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/allu/CableReportServiceAlluITests.kt

```diff
@@ -68,6 +70,45 @@ class CableReportServiceAlluITests {
         assertThat(request.getHeader("Authorization")).isEqualTo("Bearer $stubbedBearer")
     }
 
+    @Test
+    fun `addAttachment should upload attachment`() {
+        addStubbedLoginResponse()
+        val alluId = 123
+        val metadata = AlluDataFactory.createAttachmentMetadata()
+        val file = "test file content".toByteArray()
+        val attachment = Attachment(metadata, file)
+        val mockResponse = MockResponse().setResponseCode(200)
+        mockWebServer.enqueue(mockResponse)
+
+        service.addAttachment(alluId, attachment)
+
+        assertThat(mockWebServer.takeRequest()).isValidLoginRequest()
+        val request = mockWebServer.takeRequest()
+        assertThat(request.method).isEqualTo("POST")
+        assertThat(request.path).isEqualTo("/v2/applications/$alluId/attachments")
+    }
+
+    @Test
+    fun `addAttachments should upload attachments successfully`() {
+        addStubbedLoginResponse()
+        val alluId = 123
+        val metadata = AlluDataFactory.createAttachmentMetadata()
+        val file = "test file content".toByteArray()
+        val attachment = Attachment(metadata, file)
+        val mockResponse = MockResponse().setResponseCode(200)
+        (1..3).forEach { _ -> mockWebServer.enqueue(mockResponse) }
+        val attachments = listOf(attachment, attachment, attachment)
+
+        service.addAttachments(alluId, attachments)
+
+        assertThat(mockWebServer.takeRequest()).isValidLoginRequest()
+        assertThat(attachments).each {
```

Comment:
We're not asserting anything from attachments.
```suggestion
        attachments.foreach {
```

Reply:
Changed as suggested.